box/schema_user.go

```diff
@@ -0,0 +1,542 @@
+// It works just like hash function.
+func (u *SchemaUser) Password(ctx context.Context, password string) (string, error) {
+	// Create the request and send it to Tarantool.
+	req := NewUserPasswordRequest(password).Context(ctx)
+	resp := &UserPasswordResponse{}
+
+	// Execute the request and handle the response.
+	fut := u.conn.Do(req)
+
+	// Get the decoded response.
+	err := fut.GetTyped(resp)
+	if err != nil {
+		return "", err
+	}
+
+	// Return the password hash.
+	return resp.Hash, nil
+}
+
+// UserPasswdRequest represents a request to change a user's password in Tarantool.
+type UserPasswdRequest struct {
+	*tarantool.CallRequest // Underlying Tarantool call request.
+}
+
+// NewUserPasswdRequest creates a new request to change  a user's password in Tarantool.
+func NewUserPasswdRequest(args ...string) (UserPasswdRequest, error) {
+	callReq := tarantool.NewCallRequest("box.schema.user.passwd")
+
+	switch len(args) {
+	case 1:
+		callReq.Args([]interface{}{args[0]})
+	case 2:
+		callReq.Args([]interface{}{args[0], args[1]})
+	default:
+		return UserPasswdRequest{}, fmt.Errorf("len of fields must be 1 or 2, got %d", len(args))
+
+	}
+
+	return UserPasswdRequest{callReq}, nil
+}
+
+// UserPasswdResponse represents the response to a user passwd request.
+type UserPasswdResponse struct{}
+
+// Passwd sends a request to set a password for a currently logged in or a specified user.
+// A currently logged-in user can change their password using box.schema.user.passwd(password).
+// An administrator can change the password of another user
+// with box.schema.user.passwd(username, password).
+func (u *SchemaUser) Passwd(ctx context.Context, args ...string) error {
+	req, err := NewUserPasswdRequest(args...)
+	if err != nil {
+		return err
+	}
+
+	req.Context(ctx)
+
+	resp := &UserPasswdResponse{}
+
+	// Execute the request and handle the response.
+	fut := u.conn.Do(req)
+
+	err = fut.GetTyped(resp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UserInfoRequest represents a request to get a user's info in Tarantool.
+type UserInfoRequest struct {
+	*tarantool.CallRequest // Underlying Tarantool call request.
+}
+
+// NewUserInfoRequest creates a new request to get user privileges.
+func NewUserInfoRequest(username string) UserInfoRequest {
+	callReq := tarantool.NewCallRequest("box.schema.user.info").Args([]interface{}{username})
+
+	return UserInfoRequest{
+		callReq,
+	}
+}
+
+// PrivilegeObjectType is a struct based on privilege object types list
+// https://www.tarantool.io/en/doc/latest/admin/access_control/#all-object-types-and-permissions
+type PrivilegeObjectType string
+
+const (
+	// PrivilegeObjectTypeUniverse - privilege type based on universe.
+	// A database (box.schema) that contains database objects, including spaces,
+	// indexes, users, roles, sequences, and functions.
+	// Granting privileges to universe gives a user access to any object in the database.
+	PrivilegeObjectTypeUniverse PrivilegeObjectType = "universe"
```

Comment:
Can we use this naming?

```suggestion
// https://www.tarantool.io/en/doc/latest/admin/access_control/#all-object-types-and-permissions
type PrivilegeType string

const (
	// PrivilegeUniverse - privilege type based on universe.
	// A database (box.schema) that contains database objects, including spaces,
	// indexes, users, roles, sequences, and functions.
	// Granting privileges to universe gives a user access to any object in the database.
	PrivilegeUniverse PrivilegeType = "universe"
```

Reply:
fixed